pkg/util/constants.go

```diff
@@ -238,6 +249,21 @@ const (
 var (
 	// EndpointSliceGVK is the GroupVersionKind of K8s native EndpointSlice.
 	EndpointSliceGVK = discoveryv1.SchemeGroupVersion.WithKind("EndpointSlice")
+
+	// ManagedByKarmadaLabels is map of reserved karmada label to indicate whether resources are workloads
+	// managed by karmada controllers.
+	ManagedByKarmadaLabels = map[string]string{
+		ManagedByKarmadaLabel: ManagedByKarmadaLabelValue,
+	}
+
+	// KarmadaSystemLabels is map of reserved karmada label to indicate whether resources are system level
+	// resources managed by karmada controllers.
+	KarmadaSystemLabels = map[string]string{
+		KarmadaSystemLabel: KarmadaSystemLabelValue,
+	}
+
+	// KarmadaAgentLabels is map of reserved karmada label for karmada-agent deployment.
+	KarmadaAgentLabels = map[string]string{"app": KarmadaAgentName}
```

Comment:
Implemented per this comment from five months ago: https://github.com/karmada-io/karmada/pull/4620#discussion_r1493130385

Happy to revert to the way it was

Reply:
I suggest using the way it was before the PR was submitted.

Reply:
so to be clear:

re: `karmadaAgentLabels` was a map before the PR was submitted so leave as-is: https://github.com/karmada-io/karmada/blob/71de3dc02247e22da249c285fa0c0c032ae01445/pkg/karmadactl/register/register.go#L115

re: `KarmadaManagedLabels` and disregard @zhzhuang-zju suggestion and create maps inline as opposed to reusing one?

re: `KarmadaSystemLabels` follow managed labels

Reply:
@a7i @XiShanYongYe-Chang It's great to have a discussion about how to make the code better~ In fact, this question can be divided into two parts: 1. whether it is necessary to wrap the label as a map. 2. if it is wrapped as a map, whether it is necessary to export it. refer to https://github.com/karmada-io/karmada/pull/4620#discussion, I meant that if it is wrapped as a map, it is possible to export it for better scalability. I think what @XiShanYongYe-Chang means is that step 1 is not necessary because there is only one label, so there is no need to wrap it into a map. That's something I agree with, and I think the final opinion is that we can leave merged code alone, and the new code does not need to wrap the label into a map again. What do you think?

Reply:
Can you please be more specific with the three labels to avoid back and forth?

Reply:
Just make it keep with before will be ok, just like this:
```go
var karmadaAgentLabels = map[string]string{"app": KarmadaAgentName}

func XXX(){
       ...
        ObjectMeta: metav1.ObjectMeta{
	        Name:      KarmadaAgentName,
	        Namespace: o.Namespace,
        	Labels:    karmadaAgentLabels,
        },
       ...
}
```

Reply:
combine with https://github.com/karmada-io/karmada/pull/4620#discussion_r1673916240 and https://github.com/karmada-io/karmada/pull/4620#discussion_r1673921429 is ok. @a7i ,If we didn't make it clear, please let us know

Reply:
>just replaced with labelName: labelvalue wherever used.

It's not very DRY, but at this point it's been 5 months and I'd like to get the PR moving. Commit coming up soon

Reply:
I hope I got it right this time 🤞🏼 

ready for your review